src/pages/OriginalSampleRegistration.tsx

```diff
@@ -146,15 +115,124 @@ function OriginalSampleRegistration({ registrationInfo }: RegistrationParams) {
+              sampleCollectionDate: tissue.sampleCollectionDate
+                ? tissue.sampleCollectionDate instanceof Date
+                  ? tissue.sampleCollectionDate.toLocaleDateString()
+                  : tissue.sampleCollectionDate
+                : undefined
+            };
+
+            return sampleRegisterData;
+          })
+        ];
+      }, []);
+      setSampleRegistrationData(samples);
+      return new Promise((resolve) => {
+        resolve({ request: { samples } });
+      });
+    },
+    [setSampleRegistrationData]
+  );
+
+  const resultColumns: Array<Column<LabwareResultData>> = React.useMemo(() => {
+    return [
+      {
+        Header: 'Barcode',
+        id: 'barcode',
+        accessor: (result: LabwareResultData) => result.labware.barcode
+      },
+      {
+        Header: 'Labware Type',
+        id: 'labwareType',
+        accessor: (result: LabwareResultData) => result.labware.labwareType.name
+      },
+      {
+        Header: 'External ID',
+        id: 'externalID',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(result.labware, (sample: SampleFieldsFragment) => sample.tissue.externalName ?? '')
+      },
+      {
+        Header: 'Donor ID',
+        id: 'donorId',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(result.labware, (sample: SampleFieldsFragment) => sample.tissue.donor.donorName ?? '')
+      },
+      {
+        Header: 'Tissue type',
+        id: 'tissueType',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(
+            result.labware,
+            (sample: SampleFieldsFragment) => sample.tissue.spatialLocation.tissueType.name ?? ''
+          )
+      },
+      {
+        Header: 'Spatial location',
+        id: 'spatialLocation',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(result.labware, (sample: SampleFieldsFragment) => String(sample.tissue.spatialLocation.code))
+      },
+      {
+        Header: 'Replicate',
+        id: 'replicate',
+        accessor: (result: LabwareResultData) =>
+          valueFromSamples(result.labware, (sample: SampleFieldsFragment) => String(sample.tissue.replicate))
+      },
+      {
+        Header: 'Fixative',
+        id: 'fixative',
+        accessor: (result: LabwareResultData) =>
+          result.labware.slots[0].samples.length > 0 ? result.labware.slots[0].samples[0].tissue.fixative.name : ''
+      },
+      {
+        Header: 'Solution',
+        id: 'solution',
+        accessor: (result: LabwareResultData) =>
+          result.extraData && result.extraData.length > 0 ? result.extraData[0] : ''
+      }
+    ];
+  }, []);
+
+  const formatSuccessData = React.useCallback(
+    (labware: LabwareFieldsFragment[]) => {
+      return labware.map((lw, indx) => {
+        return {
+          labware: lw,
+          extraData:
+            sampleRegistrationData && sampleRegistrationData.length > indx
```

Comment:
not 100% on what this is doing. Is extra data only ever the solution from sampleRegistrationData? If so does it need to be called extraData, could just be called solution. Also it could be flakey grabbing a value based on a position in an array, is there a better way to link labware to the sampleRegistration data?

Reply:
As it is template structure, I thought that will give flexibility to provide any other extra data that we need to disaply other than labware, if such cases arise in future realted to other registrations as well.I know, the ideal would have been to return solution as part of mutation result. But another

Reply:
Ok thats fair enough. Im not too sure on this use of index though. It relies on the labware and sampleRegistration being the same length and order

Reply:
Yes, I have asked David whether it is possible to return the 'solution' as part of the mutation result from which we are getting labware information.

Reply:
Ah ok sorry I misunderstood 👍

Reply:
It's ok. Again, do you an any other appropriate name for 'extraData' field?

Reply:
I think extraData is probably fine because its a broad range of data so its not specific. Maybe registrationData or something but up to you.